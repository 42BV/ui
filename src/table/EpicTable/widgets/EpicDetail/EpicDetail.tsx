import { Icon } from '../../../../core/Icon';
import { t } from '../../../../utilities/translation/translation';
import { ReactNode } from 'react';

type Text = {
  /**
   * The text to show next to the close button.
   */
  close?: string;
};

type Props = {
  /**
   * The details you want to render.
   */
  children: ReactNode;

  /**
   * The callback for the close event. Is called when the close button
   * is clicked.
   */
  onClose: (event: MouseEvent<HTMLElement>) => void;

  /**
   * Optionally customized text within the component.
   * This text should already be translated.
   */
  text?: Text;
};

/**
 * EpicDetail is a white pane with a close button which can be used
 * to render in a EpicDetailRow as its content.
 */
export function EpicDetail({ children, onClose, text = {} }: Props) {
  return (
    <div className="bg-white h-100 border-top border-end border-bottom py-1 shadow">
      <div className="d-flex align-items-center border-bottom">
        <Icon className="text-muted py-2 px-1" onClick={onClose} icon="close" />
        <span className="text-muted">
          {t({
            key: 'EpicTable.EpicDetail.CLOSE',
            fallback: 'Close',
            overrideText: text.close
          })}
        </span>
      </div>

      <div className="p-2">{children}</div>
    </div>
  );
}
