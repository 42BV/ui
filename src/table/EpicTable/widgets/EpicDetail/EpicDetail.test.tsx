import React from 'react';
import { render } from '@testing-library/react';

import { EpicDetail } from './EpicDetail';
import { fireEvent, screen } from '@testing-library/react';

describe('Component: EpicDetail', () => {
  function setup() {
    const onCloseSpy = jest.fn();


    const { container } = render(
      <EpicDetail onClose={onCloseSpy}>
        <h1>children</h1>
      </EpicDetail>
    );

    return { container, onCloseSpy };
  }

  test('ui', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  describe('events', () => {
    it('should call the onClose callback when the close button is clicked', () => {
      const { onCloseSpy } = setup();

      fireEvent.click(screen.getByText('close'));

      expect(onCloseSpy).toBeCalledTimes(1);
    });
  });
});
