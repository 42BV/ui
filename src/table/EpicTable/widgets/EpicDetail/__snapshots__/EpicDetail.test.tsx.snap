// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: EpicDetail ui 1`] = `
<div
  className="bg-white h-100 border-top border-right border-bottom py-1 shadow"
>
  <div
    className="d-flex align-items-center border-bottom"
  >
    <Icon
      className="text-muted py-2 px-1"
      icon="close"
      onClick={[MockFunction]}
    />
    <span
      className="text-muted"
    >
      Close
    </span>
  </div>
  <div
    className="p-2"
  >
    <Component />
  </div>
</div>
`;
